core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -204,7 +203,7 @@ public void copyLogSegmentData(final RemoteLogSegmentMetadata remoteLogSegmentMe
                 final InputStream transactionIndex = Files.newInputStream(logSegmentData.transactionIndex().get());
                 uploadIndexFile(remoteLogSegmentMetadata, transactionIndex, TRANSACTION);
             }
-            final ByteBufferInputStream leaderEpoch = new ByteBufferInputStream(logSegmentData.leaderEpochIndex());
+            final InputStream leaderEpoch = new ByteArrayInputStream(logSegmentData.leaderEpochIndex().array());
```

Comment:
Is there any particular reason for this change?

Reply:
Good question. Metric tests were failing with ByteBuffer input stream, and though there may be some issue with the state of the byte buffer state received at this point.
Looking deeper though, it was the source byteBuffer causing this issue. So by flipping it after each copy in the Metrics test it's passing correctly. 

So, rolling back this change, to avoid copying byte arrays. Also checked that resulting leader epoch is correctly uploaded. Pushing changes now.